docs/02-for-app-authors/03-appdata-guidelines/index.md

```diff
@@ -107,36 +160,140 @@ The old desktop file name is automatically added if we use `rename-desktop-file`
 </provides>
 ```
 
-## Icons and categories
+Console applications must add their main binary name to the provides tag:
 
-If there’s a `type="desktop-id"` [launchable](#launchable), they get pulled from it. Most of the icon not found errors with the flathub builder can be traced down to the `launchable` value not matching the desktop file name.
+```xml
+<provides>
+  <binary>foo</binary>
+</provides>
+```
+
+Please see the [specification](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-provides)
+to know more.
+
+## Categories and keywords
+
+If there’s a `type="desktop-id"` [launchable](#launchable), they are 
+pulled from the desktop file and merged in the appdata. So defining them 
+seperately in the appdata is not necessary.
+
+Please see the [specification](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-categories) 
+on how to define them in appdata.
 
-Don’t set them in the AppData unless you want to override them (even though then it might be a better idea to patch the desktop file itself).
+If they are present in both places, appstream will merge them.
+
+Please don't use any of the following categories: 
+`KDE, GTK, Qt, Application, GNOME, GUI`.
```

Comment:
I'm actually unsure, why we recommend against this. While I agree, these shouldn't be categories, we don't have something better (keywords are not that good for this IMO).

And as far as I've seen, some power users actually would want to look for apps from a technical/framework POV.